core/src/main/scala/org/bitcoins/core/protocol/ln/LnCurrencyUnit.scala

```diff
@@ -0,0 +1,239 @@
+
+  def apply(nanoBitcoins: Int64): NanoBitcoins = NanoBitcoins(nanoBitcoins.toBigInt)
+
+  def apply(underlying: BigInt): NanoBitcoins = NanoBitcoinsImpl(underlying)
+
+  private case class NanoBitcoinsImpl(underlying: BigInt) extends NanoBitcoins {
+    require(underlying <= maximum, "Number was too big for NanoBitcoins, got: " + underlying)
+    require(underlying >= minimum, "Number was too small for NanoBitcoins, got: " + underlying)
+  }
+}
+
+sealed abstract class PicoBitcoins extends LnCurrencyUnit {
+  override type A = BigInt
+
+  override def character: Char = 'p'
+
+  override def multiplier: BigDecimal = 0.000000000001
+
+  override def toBigInt: A = underlying
+
+  override def toSatoshis: Satoshis = LnCurrencyUnits.toSatoshi(this)
+
+  override def toPicoBitcoins: BigInt = this.toBigInt
+}
+
+object PicoBitcoins extends BaseNumbers[PicoBitcoins] {
+  val min = PicoBitcoins(minimum)
+  val max = PicoBitcoins(maximum)
+  val zero = PicoBitcoins(0)
+  val one = PicoBitcoins(1)
+
+  def minimum: BigInt = -(Consensus.maxMoney.toBigDecimal / (Bitcoins.one.satoshis.toBigDecimal * 0.000000000001)).toBigInt() //TODO: Replace with LnCurrencyUnits.PicoMultiplier
+
+  def maximum: BigInt = (Consensus.maxMoney.toBigDecimal / (Bitcoins.one.satoshis.toBigDecimal * 0.000000000001)).toBigInt() //TODO: Replace with LnCurrencyUnits.PicoMultiplier
+
+  def apply(picoBitcoins: Int64): PicoBitcoins = PicoBitcoins(picoBitcoins.toBigInt)
+
+  def apply(underlying: BigInt): PicoBitcoins = PicoBitcoinsImpl(underlying)
+
+  private case class PicoBitcoinsImpl(underlying: BigInt) extends PicoBitcoins {
+    require(underlying <= maximum, "Number was too big for PicoBitcoins, got: " + underlying)
+    require(underlying >= minimum, "Number was too small for PicoBitcoins, got: " + underlying)
+  }
+}
+
+object LnCurrencyUnits {
+  val oneMilliBTC: LnCurrencyUnit = MilliBitcoins.one
+  val oneMicroBTC: LnCurrencyUnit = MicroBitcoins.one
+  val oneNanoBTC: LnCurrencyUnit = NanoBitcoins.one
+  val onePicoBTC: LnCurrencyUnit = PicoBitcoins.one
+
+  val MilliMultiplier: BigDecimal = BigDecimal(0.001)
+  val MicroMultiplier: BigDecimal = BigDecimal(0.000001)
+  val NanoMultiplier: BigDecimal = BigDecimal(0.000000001)
+  val PicoMultiplier: BigDecimal = BigDecimal(0.000000000001)
+
+  def apply(lnUnit: LnCurrencyUnit, amount: BigInt): LnCurrencyUnit = lnUnit match {
+    case b: MilliBitcoins => MilliBitcoins(amount)
+    case c: MicroBitcoins => MicroBitcoins(amount)
+    case d: NanoBitcoins => NanoBitcoins(amount)
+    case e: PicoBitcoins => PicoBitcoins(amount)
+  }
+
+  def getZeroUnit(lnUnit: LnCurrencyUnit): LnCurrencyUnit = apply(lnUnit, BigInt(0))
+
+  def toPicoBitcoins(lnCurrencyUnits: LnCurrencyUnit): BigInt = lnCurrencyUnits.toBigInt * (lnCurrencyUnits.multiplier / LnCurrencyUnits.PicoMultiplier).toBigInt()
```

Comment:
That is because what I really wanted was to get the PicoBitcoin value, not necessarily an actual PicoBitcoin object. I can rename the function, or return an object, either will work. I will think a bit more about which would be better.